presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1133,7 +1133,16 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
                 tableProperties);
 
         WriteInfo writeInfo = locationService.getQueryWriteInfo(locationHandle);
-        metastore.declareIntentionToWrite(session, writeInfo.getWriteMode(), writeInfo.getWritePath(), result.getFilePrefix(), schemaTableName, false);
+        boolean sorted = bucketProperty.filter(property -> !property.getSortedBy().isEmpty()).isPresent();
```

Comment:
This is potentially error prone. What if some other feature uses the tmp path to store whatsoever? What are the implications of dropping the tmp folders unconditionally? 

Do you think we shouldn't request the `tmpPath` from the LocationService if it is not needed? Then we can also pre-created it in the location service setting all the required permissions (similarly to writePath)?

Reply:
This is a good thought, let me see how we could convey the `tmpPath` request.